### PR TITLE
ENG-6606: Introduce event cursor support

### PR DESCRIPTION
**Pull request by erickpintor**
### Description

Provide support for the new per-event stream cursor.

### Motivation and context

There's an edge case where resuming a stream from a txn time with lots of documents written could result in event loss. This approach fixes that edge case.

### How was the change tested?

Unit tests included.

### Screenshots (if appropriate):

NA

### Change types

* - [X] Bug fix (non-breaking change that fixes an issue)
* - [X] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [X] My code follows the code style of this project.
* - [X] My change requires a change to Fauna documentation.
* - [X] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -636,6 +645,8 @@ export class StreamClient<T extends QueryValue = any> {
   #query: () => Promise<StreamToken>;
   /** The last `txn_ts` value received from events */
   #last_ts?: number;
+  /** The last `cursor` value received from events */
+  #last_cursor?: string;
```
Comment (ptpaterson) [inline src/client.ts:649]: `cursor` was added as an option, but this is private so folks won't be able to pick it out of the stream client after an error. I think we can make this easier.

Options:
1. Do nothing: folks can store their own copy of the cursor as events come in, it's just an extra thing they have to remember to do.
2. Add a getter method to read the `#last_cursor`. Then folks could read it from a closed StreamClient and create a new one with the `cursor` option.
3. Remove the `cursor` option and provide a `restart` method that would establish a new connection with the stored `#last_cursor`.

Letting folks configure a cursor option is the most flexible, but if we want to do that, then I think we should make it easier to get the cursor after an error. Having a `restart` method would be one of the easiest things for a customer to use, but maybe not as flexible.

**Review by ptpaterson — CHANGES_REQUESTED** [review]
Nice. This is looking good! Just some thoughts on using the cursor.

File: src/client.ts (hunk 1 above)
Comment (erickpintor) [reply src/client.ts:649]: The intent is to be able to recover from a crash, in which case you can't rely on the `last_cursor` as memory would be lost in that case so, I feel like the `restart` would come short in that regard. I'll add a getter since that's pretty harmless but, in the case of crash recovery, users can't rely on that either.

Comment (erickpintor) [reply src/client.ts:649]: You gotta persist it in durable storage (disk or somewhere else).

Comment (pnwpedro) [reply src/client.ts:649]: You need to persist it as you handle an event if you want to recover and this impl supports that, so I'm not sure adding a getter is actually harmless. It can throw devs off. They shouldn't implement cursor persistence by polling the client, for example.

Comment (ptpaterson) [reply src/client.ts:649]: Okay, thanks! I concur with Lucas about not needing the getter.

**Review by ptpaterson — APPROVED** [review]
